Adafruit_GFX.cpp

```diff
@@ -1148,27 +1152,36 @@ void Adafruit_GFX::drawChar(int16_t x, int16_t y, unsigned char c,
 
     startWrite();
     for (int8_t i = 0; i < 5; i++) { // Char bitmap = 5 columns
-      uint8_t line = pgm_read_byte(&font[c * 5 + i]);
-      for (int8_t j = 0; j < 8; j++, line >>= 1) {
-        if (line & 1) {
-          if (size_x == 1 && size_y == 1)
-            writePixel(x + i, y + j, color);
-          else
-            writeFillRect(x + i * size_x, y + j * size_y, size_x, size_y,
-                          color);
-        } else if (bg != color) {
-          if (size_x == 1 && size_y == 1)
-            writePixel(x + i, y + j, bg);
-          else
-            writeFillRect(x + i * size_x, y + j * size_y, size_x, size_y, bg);
+      uint8_t count = 0;
+      uint8_t j_new, j, line = pgm_read_byte(&font[c * 5 + i]);
+      j = j_new = 0;
+      while (j < 8) {
```

Comment:
Thanks for your suggestion; I'll take another look.

I think you misinterpreted the point of this new code. Its effect on the 1-bpp displays is mostly irrelevant because Adafruit's GFX library always draws into RAM. The point of these changes is for **color SPI displays** like the ILI9341 where every single pixel operation is a large expense in time. Please retest on a color display.